packages/runtime/src/next/client.ts

```diff
@@ -201,23 +230,51 @@ export class Makeswift {
     return response
   }
 
-  async getPages({
-    siteVersion = MakeswiftSiteVersion.Live,
-  }: {
-    siteVersion?: MakeswiftSiteVersion
-  } = {}): Promise<MakeswiftPage[]> {
-    const response = await this.fetch(`v3/pages`, siteVersion)
+  private getPagesInternal = async (
+    args: {
+      siteVersion?: MakeswiftSiteVersion
+      limit?: number
+      after?: string
+      sortBy?: 'title' | 'path' | 'createdAt' | 'updatedAt' | 'publishedAt'
+      sortDirection?: 'asc' | 'desc'
+      includeOffline?: boolean
+      pathPrefix?: string
+      locale?: string
+    } = {},
+  ) => {
+    const {
+      siteVersion = MakeswiftSiteVersion.Live,
+      limit = 20,
+      after,
+      includeOffline,
+      sortBy,
+      sortDirection,
+      pathPrefix,
+      locale,
+    } = args
```

Comment:
Why aren't we destructuring arguments directly in the arguments? Do we need this intermediate `args` variable?

Reply:
If we factor out the query construction out of this method, we wouldn't need to destructure them at all here. That said, IIRC, we originally had this argument as optional, and you can't destructure an optional arg.

Reply:
Ah, makes sense. As for destructuring from an optional param, this is what I usually do:

```ts
class Foo() {
  someMethod({ foo = 1, bar = '', baz }: { foo?: number, baz?: string, baz?: string } = {}) {
    //
  }
}
```

If not all fields were optional, then it wouldn't make sense for the parameter itself to be optional so this pattern works in all cases where you have optional named parameters.